docs/debugging.asciidoc

```diff
@@ -0,0 +1,23 @@
+[[debugging]]
+== Debugging the agent itself
+
+Hopefully the agent Just Works™ for you but depending on your situation the agent could need some tuning.
+
+Firstly, to know more about what's going on inside the agent, you can increase the amount of log messages it writes by setting the log level with the option `log_level = 0` -- `0` being the level of most messages, `DEBUG`.
+
+[float]
+[[debugging-errors]]
+=== Errors
+
+[float]
+[[debugging-errors-queue-full]]
+==== `Queue is full (256 items), skipping…`
+
+The agent has an internal queue that holds events after they are done, before they are safely serialized and sent to APM Server. To avoid using up all of your memory, the queue has a fixed size. Taking from the queue happens faster than adding to it but it can happen that the queue fills, hence the warning.
```

Comment:
> Taking from the queue happens faster than adding to it but it can happen that the queue fills

Aren't those two statements contradictory? If taking always happens faster, then the queue cannot fill. Whether it's typical or not really comes down to the application, and what level of load is normal.

Perhaps restate like "Depending on the load of your application, events may be added to the queue faster than they can be consumed ..."

Reply:
I think I just missed a "_usually faster_"